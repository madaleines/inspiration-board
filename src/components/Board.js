import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';
import CARD_DATA from '../data/card-data.json';

class Board extends Component {
  constructor() {
    super();
    this.state = {
      cards: [],
    };
  }

  renderCardComponents = () => {
    const cards = CARD_DATA.cards.map((card, index) => {
      return(
        <Card
          key={index}
          text={card.text}
          emoji={card.emoji}
          />
      )
    })
    return cards;
  }

  render() {
    const renderedCards = this.renderCardComponents()
    return (
      <div>
        { renderedCards }
      </div>
    )
  }

}

Board.propTypes = {

};

export default Board;
