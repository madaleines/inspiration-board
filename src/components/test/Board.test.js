import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Board from '../Board.js';

describe('Board', () => {
  test('that it renders Board with shallow rendering', () => {
    const wrapper = shallow(<Board url="url"
    boardName="boardName"/>);

    expect(wrapper).toMatchSnapshot();

    // wrapper.unmount();
  });

});
