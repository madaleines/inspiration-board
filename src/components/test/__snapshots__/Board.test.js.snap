// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board that it renders Board with shallow rendering 1`] = `<div />`;
